addons/hr_holidays/models/resource.py

```diff
@@ -157,6 +160,57 @@ def unlink(self):
 
         return res
 
+    @api.model
+    def load_public_holidays(self, companies=[], from_ui=False):
+        company_ids = self.env['res.company'].browse(companies)
+        if not company_ids:
+            return
+
+        current_year = datetime.now().year
+        existing_holidays = self.env["resource.calendar.leaves"].search([
+            ('company_id', '=', company_ids.ids),
+            ('resource_id', '=', False),
+        ])
+        for company in company_ids:
+            if not company.country_code:
+                continue
+            user_tz = pytz.timezone(company.resource_calendar_id.tz)
+            public_holiday_values = []
+            public_holiday_dict = holidays.country_holidays(company.country_code, years=range(current_year, current_year + 4))
+
+            for holiday_date, holiday_name in public_holiday_dict.items():
+                if from_ui:
+                    holiday_start_utc = user_tz.localize(
+                        datetime.combine(holiday_date, time.min)).astimezone(pytz.utc).replace(tzinfo=None)
+                    holiday_end_utc = user_tz.localize(
+                        datetime.combine(holiday_date, time.max)).astimezone(pytz.utc).replace(tzinfo=None)
+                else:
+                    holiday_start_utc = datetime.combine(holiday_date, time.min).astimezone(pytz.utc).replace(tzinfo=None)
+                    holiday_end_utc = datetime.combine(holiday_date, time.max).astimezone(pytz.utc).replace(tzinfo=None)
+                present_in_holiday_vals = any(
+                    holiday['date_from'] == holiday_start_utc and holiday['company_id'] == company.id
+                        for holiday in public_holiday_values
+                    )
+                overlapping_holidays = existing_holidays.filtered(lambda leave:
+                    leave.company_id == company and
```

Comment:
Though there is a default, the field isn't required
```suggestion
                    (leave.company_id == company or not leave.company_id) and
```